src/main/java/cloud/fogbow/rcs/core/service/cache/MemoryBasedCache.java

```diff
@@ -0,0 +1,72 @@
+package cloud.fogbow.rcs.core.service.cache;
+
+import cloud.fogbow.common.exceptions.FogbowException;
+import cloud.fogbow.rcs.core.PropertiesHolder;
+import org.apache.commons.lang.time.DateUtils;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static cloud.fogbow.rcs.constants.ConfigurationPropertyDefaults.CACHE_EXPIRATION_TIME_DEFAULT;
+import static cloud.fogbow.rcs.constants.ConfigurationPropertyKeys.CACHE_EXPIRATION_TIME_KEY;
+
+public class MemoryBasedCache<T> implements CacheService<T> {
+    private final int CACHE_VALIDITY;
```

Comment:
Aren't constants immutable? This looks more like a variable or attribute... Make sure it's correct.

Reply:
I can make it constant and initialize just in the class constructor so that it becomes immutable from now on inside the class.